titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/Version.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.api.jobmanager.model.job;
+
+import java.util.Objects;
+
+/**
+ * TODO Add monotonically increasing `revision` number unique within a cell
```

Comment:
Would this revision number replace the timestamp or be in addition to? Seems like they would be very similar.

Reply:
It would be in addition to it. Timestamp has a meaning and allows us to measure processing latencies. Revision id is needed for strict ordering and implementation of the consistent caching in gateway.